docs/xss.rst

```diff
@@ -0,0 +1,96 @@
+Performing XSS Vulnerability Audits
+===================================
+
+The bok-choy framework includes the ability to perform XSS (cross-site scripting) audits on
+web pages using a short XSS locator defined in
+https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#XSS_Locator.
+
+Trigger XSS Vulnerability Audits in Existing Tests
+--------------------------------------------------
+
+You might already have some bok-choy tests written for your web application. To
+leverage existing bok-choy tests and have them fail on finding XSS vulnerabilities,
+follow these steps.
+
+1. Insert the ``XSS_INJECTION`` string defined in ``bok_choy.page_object`` into your page content.
+2. Set the ``VERIFY_XSS`` environment variable to ``True``.
+
+::
+
+    export VERIFY_XSS=True
+
+
+An XSS audit is triggered whenever a page object's ``q`` method is called. The audit will detect
+improper escaping both in HTML and in Javascript that is embedded within HTML.
+
+If errors are found on the page, an XSSExposureError is raised.
+
+Here is an example of a bok-choy test that will implicitly check for XSS vulnerabilities.
+It clicks a button on the page and assumes that the user's name is
+displayed on the page. If the user name is not properly escaped, the display
+of the name (which is data provided by the user and thus potentially malicious) can cause
+XSS issues.
+
+In the case of the ``test_button_click_output`` test case in the example below,
+an audit will be done in the ``click_button()`` ``output()`` and method calls,
+as each of those will call out to ``q``.
+
+If any XSS errors are found, then the test case will fail with an
+XSSExposureError.
+
+.. code-block:: python
+
+    from bok_choy.page_object import PageObject, XSS_INJECTION
+
+
+    class MyPage(PageObject):
+        def url(self):
+            return 'https://www.mysite.com/page'
+
+        def is_browser_on_page(self):
+            return self.q(css='div#fixture button').present
+
+        def click_button(self):
+            """
+            Click on the button element (id="button").
+            On my example page this will trigger an ajax call
+            that updates the #output div with the text "yes!"
+            """
+            self.q(css='div#fixture button').first.click()
+            self.wait_for_ajax()
+
+        @property
+        def output(self):
+            """
+            Return the contents of the "#output" div on the page.
+            """
```

Comment:
nit: I had to read through the code twice to understand why the assertion was looking for the output to be 'yes!'.
Maybe it would have been more clear the first time through if this docstring also said something like:

> In the example page, it will contain the text "yes!" after being updated by the ajax call that is triggered by clicking the button.

This might be overkill, but I'd rather over-explain everything in the docs to account for new coders, non-native English speakers, etc.